### PR TITLE
Fix Windows exception handling for newer LLVM interfaces.

### PR DESCRIPTION
**Pull request by marsupial**
Block it’s usage around CLING_WIN_SEH_EXCEPTIONS macro.

### CODE REVIEW

File: lib/Utils/PlatformWin.cpp [hunk 1]
```diff
@@ -735,49 +739,64 @@ static void MergeRanges(ImageRanges &Ranges) {
...
   }
 
-  ::RtlAddFunctionTable(RFunc, N, BaseAddr);
+  if (!Block)
+    MergeRanges(Ranges); // Initial sort and merge
 }
 
-void DeRegisterEHFrames(uint8_t* Addr, size_t Size) {
-  assert(getImageBaseMap().find(DWORD64(Addr)) != getImageBaseMap().end());
+void DeRegisterEHFrames(uintptr_t ImgBase, EHFrameInfos Frames) {
```
Comment (bellenot) [inline lib/Utils/PlatformWin.cpp]: Should be: `void DeRegisterEHFrames(uintptr_t BaseAddr, const EHFrameInfos& Frames) {`

Comment (marsupial) [inline lib/Utils/PlatformWin.cpp]: Indeed....How'd that happen?

**Comment by bellenot** [comment]
BTW, why blocking it’s usage around CLING_WIN_SEH_EXCEPTIONS macro?
Without defining CLING_WIN_SEH_EXCEPTIONS, it gives the following compilation error:
```
"C:\Users\bellenot\build\release\ALL_BUILD.vcxproj" (default target) (1) ->
"C:\Users\bellenot\build\release\tools\cling\tools\driver\cling.vcxproj" (default target) (235) ->
"C:\Users\bellenot\build\release\tools\cling\lib\Interpreter\clingInterpreter.vcxproj" (default target) (236) ->
"C:\Users\bellenot\build\release\tools\cling\lib\Interpreter\obj.clingInterpreter.vcxproj" (default target) (237) ->
(ClCompile target) ->
  C:\Users\bellenot\git\original\tools\cling\lib\Interpreter\Interpreter.cpp(467): error C2039: 'ClingRaiseSEHException
': is not a member of 'cling::utils::platform' [C:\Users\bellenot\build\release\tools\cling\lib\Interpreter\obj.clingIn
terpreter.vcxproj]
  C:\Users\bellenot\git\original\tools\cling\lib\Interpreter\Interpreter.cpp(467): error C2065: 'ClingRaiseSEHException
': undeclared identifier [C:\Users\bellenot\build\release\tools\cling\lib\Interpreter\obj.clingInterpreter.vcxproj]
  C:\Users\bellenot\git\original\tools\cling\lib\Interpreter\Interpreter.cpp(467): error C2664: 'bool cling::Incrementa
lExecutor::addSymbol(const char *,void *,bool)': cannot convert argument 2 from 'bool' to 'void *' [C:\Users\bellenot\b
uild\release\tools\cling\lib\Interpreter\obj.clingInterpreter.vcxproj]

    3 Warning(s)
    3 Error(s)
```
Then I just commented them out, or replaced with LLVM_ON_WIN32 macro, and it works...

**Comment by marsupial** [comment]
Fixed. Macro is there because:
1. I've tested this on a JIT using COFF and have no idea how/if it functions when you use ELF which master still does (and seemingly will for a while).
2. There were previous complaints that the code is 64-bit only. You can figure out how you want to enable/disable it in relation to that and the above.

**Comment by bellenot** [comment]
OK, I see, thanks for the info. It makes perfect sense

**Comment by Axel-Naumann** [comment]
Merged, thanks!

**Pull request closed without merging**